tcms/testruns/tests/test_permissions.py

```diff
@@ -128,3 +128,43 @@ def verify_post_with_permission(self):
             self.assertEqual(case.history.latest().history_id, case_run.case_text_version)
             self.assertEqual(last_run.build, case_run.build)
             self.assertEqual(None, case_run.close_date)
+
+
+class MenuAddCommentItemTestCase(PermissionsTestCase):
+    permission_label = 'django_comments.add_comment'
+    http_method_names = ['get']
+
+    @classmethod
+    def setUpTestData(cls):
+        cls.product = factories.ProductFactory(name="Kiwi")
+        cls.plan = factories.TestPlanFactory(product=cls.product)
+        cls.test_run = factories.TestRunFactory(summary='Tested summary',
+                                                plan=cls.plan)
+
+        cls.url = reverse('testruns-get', args=[cls.test_run.pk])
+        cls.post_data={'summary': 'mock',
+                       'from_plan': cls.plan.pk
+                                }
+        super().setUpTestData()
+
+        cls.add_comment_html = \
+            '<a href="#" class="addBlue9 js-show-commentdialog">{0}</a>' \
+            .format(_('Add'))
+
+    def assert_on_testrun_page(self):
+        self.assertEqual(self.test_run.summary, 'Tested summary')
+        self.assertEqual(self.test_run.plan.product.name, 'Kiwi')
```

Comment:
you should be passing the `response` object from the `verify_` methods to this method and assert that response actually contains the appropriate values displayed when we load the test runs page. ATM you are comparing if the test data matches itself which it always does (which doesn't make any sense).

you already do this (twice) in the methods below.